extensions/objects/templates/coord-two-dim-editor.component.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for coord two dim editor.
+ */
+
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { icon, latLng, LeafletEvent, LeafletMouseEvent, marker, tileLayer } from 'leaflet';
+@Component({
+  selector: 'coord-two-dim-editor',
+  templateUrl: './coord-two-dim-editor.component.html',
+  styleUrls: []
+})
+export class CoordTwoDimEditorComponent implements OnInit {
+  @Input() value;
+  @Output() valueChanged: EventEmitter<[number, number]> = new EventEmitter<
+  [number, number]>();
+  latLangValue;
+  mapEvents = {
+    map: {
+      enable: ['click'],
+      logic: 'emit'
+    },
+    markers: {
+      enable: ['dragend'],
+      logic: 'emit'
+    }
+  };
+  mapCenter = latLng(0, 0);
+  mapMarkers;
+  optionsSpec = {
+    layers: [{ url: 'http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', attribution: '© <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors' }],
```

Comment:
Should we move these strings to constants?

Reply:
Done!